reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/metrics/RollingCounter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.metrics;
+
+import java.time.Clock;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Rolling counter. The count is refreshed every interval. In every interval the count is cumulative.
+ */
+public class RollingCounter implements Counter<Long> {
+    private static final long METRICS_ROLLING_WINDOW_VALUE = 3600L;
+    private static final long METRICS_ROLLING_INTERVAL_VALUE = 60L;
+
+    private final long capacity;
+    private final long window;
+    private final long interval;
+    private final Clock clock;
+    private final ConcurrentSkipListMap<Long, Long> time2CountWin;
```

Comment:
use base class (in this case Map) for declaring unless specific methods are used

Reply:
We are using the `ConcurrentSkipListMap.headMap()` method here.